tests/vmi_cloudinit_test.go

```diff
@@ -341,8 +329,16 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 			// Expect that the secret is not present on the vmi itself
 			vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			Expect(vmi.Spec.Volumes[idx].CloudInitNoCloud.UserData).To(BeEmpty())
-			Expect(vmi.Spec.Volumes[idx].CloudInitNoCloud.UserDataBase64).To(BeEmpty())
+			foundVolume := false
+			for _, volume := range vmi.Spec.Volumes {
+				if volume.CloudInitNoCloud != nil {
+					Expect(volume.CloudInitNoCloud.UserData).To(BeEmpty())
+					Expect(volume.CloudInitNoCloud.UserDataBase64).To(BeEmpty())
+					foundVolume = true
+					break
+				}
+			}
+			Expect(foundVolume).To(BeTrue(), "couldn't find any userData volume")
```

Comment:
This looks like you took the mess from before and moved it to the bottom with a little twist.

How about this:
```go
volume := lookupCoudInitNoCloudVolume(vmi.Spec.Volumes)
Expect(volume).To(...)
```
You can either check specific fields or the whole thing.

But I really do not understand what this test is doing.
It makes no sense to me.
Why would it have explcit `UserData*` if no one inputted one? It basically checks nothing.

If the intention was to check that nothing has been modified on the spec (I really hope that is the case), then the list of volumes can be compared with the previous one and that is it. I.e. that nothing mutated them.

Reply:
It says:
```go
// Expect that the secret is not present on the vmi itself
```

I don't understand either why it would ever happens. Maybe an old bug?

Anyway, I'm not sure changing the test logic should be in the scope of this PR.

Reply:
You can send a PR to adjust it now that it was detected.
I think our work to clean the code is aimed at detecting odd things like this (after the cleanup, things are surfacing and are visible).

Anyway, if you prefer not to do this now, then can we drop the changes here? I am just unsure why the new code is better than the prev. You could use the suggested `lookup*`, at least then it may really look better than before.

Reply:
done

Reply:
Done